controller/lifecycle/lifecycle.go

```
@@ -185,8 +264,10 @@ func (l *LifecycleManager) reconcileSubroutine(ctx context.Context, instance Run
 	if instance.GetDeletionTimestamp() != nil {
 		if containsFinalizer(instance, subroutine.Finalizers()) {
 			result, err = subroutine.Finalize(ctx, instance)
-			// Remove finalizers unless requeue is requested
-			err = l.removeFinalizerIfNeeded(ctx, instance, subroutine, err, result)
```

👍 nice catch, did not see that the error was ignored